images/techdocs/README.md

```diff
@@ -21,17 +21,20 @@ services:
       target: techdocs
     working_dir: /content
     environment:
-      GOOGLE_APPLICATION_CREDENTIALS: ${GOOGLE_APPLICATION_CREDENTIALS:-}
       GCLOUD_PROJECT: ${GCLOUD_PROJECT:-}
     volumes:
       - .:/content
       - ${XDG_CACHE_HOME:-xdg-cache-home}:/root/.cache
-      - $HOME/.config/gcloud:/root/.config/gcloud
-      - ${GOOGLE_APPLICATION_CREDENTIALS:-nothing}:${GOOGLE_APPLICATION_CREDENTIALS:-/tmp/empty-GOOGLE_APPLICATION_CREDENTIALS}
+    secrets:
+      - source: google-application-credentials
+        target: /root/.config/gcloud/application_default_credentials.json
     ports:
       - "127.0.0.1:3000:3000/tcp"
       - "127.0.0.1:8000:8000/tcp"
     command: serve
+secrets:
+  google-application-credentials:
+    file: $GOOGLE_APPLICATION_CREDENTIALS
```

Comment:
Something to consider, most people do not have `GOOGLE_APPLICATION_CREDENTIALS` set. I don't. And in cases where people are running this from their computers we rather want to use have fallback to using `~/.config/gcloud`. Our aim is to have it as minimally imposing on users as possible. This may work in CI, but with this change everyone will have to have application default credentials running to be able to use it from their computers.

Maybe it is okay, but it is a tradeoff to consider. At the very least we should make some error message `${GOOGLE_APPLICATION_CREDENTIALS:?You need to have GOOGLE_APPLICATION_CREDENTIALS set}`

Reply:
I'm somewhat okay with it, but we should just be aware of the consequences.